From patchwork Tue Sep  4 22:20:21 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10588087
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 95AB0174C
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Sep 2018 22:20:32 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 8652F29F46
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Sep 2018 22:20:32 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 7986529F52; Tue,  4 Sep 2018 22:20:32 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 25D2E29F46
	for <patchwork-kvm@patchwork.kernel.org>;
 Tue,  4 Sep 2018 22:20:31 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726234AbeIECrh (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Tue, 4 Sep 2018 22:47:37 -0400
Received: from mail-pf1-f193.google.com ([209.85.210.193]:41176 "EHLO
        mail-pf1-f193.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1725825AbeIECrh (ORCPT <rfc822;kvm@vger.kernel.org>);
        Tue, 4 Sep 2018 22:47:37 -0400
Received: by mail-pf1-f193.google.com with SMTP id h79-v6so2371220pfk.8
        for <kvm@vger.kernel.org>; Tue, 04 Sep 2018 15:20:28 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=SEuS6JI6aW/LLu4usY7azA9+zhY7DvEBS6rfEZFsjmo=;
        b=oUGFb+sgm7tlrJI/6KbgMsKujTRMXo1D5mlSEythGyVnmywraxyOSXJKuTbfiiHeOT
         ACIdTotBRgf1Wst0ZMiRu2xDRa+JTTB76l8NRK3I0A75siJBU6m9uw3t9p4yXiGDDRMj
         KXTb3FogHPbZb4bQa8ipuigatNMrvBd/o6TJuxAFp41Oiqaizw2EnB3u9E5HHzJtdp8a
         B6nFPFybIcapLM6BkEg5yqBQviNnM1//5IsOFpQ0t2QpA5jIXIij0qeFer3IdxdQJ+jl
         OXfMqkP835A+RNWJ349nzGZvfyy1WNPX+PWjXJoCbT5WPnJN/p5YlkV/hYihnphkkJa+
         nu8w==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=SEuS6JI6aW/LLu4usY7azA9+zhY7DvEBS6rfEZFsjmo=;
        b=gcvimhbERjTMQ1hLFSjvVBstA65OrIeuhiO13mM08lhoYLyFPVgm61Lgo4FKhe1UXL
         gVMr4rC3f4/pwrrCN418Q5okrTrS92X03TFaJv1apiGxESpwR5Ip1R64FFcB/JOwI6Sy
         mlfpa0QEMGWL5+b8nWafJnBf8UDc7Dc70nTg+uHf6fKKzkjmCDqmMtHWpqpuSnLQQCit
         Y3bGofXvrwJg/8HnskYa/7+F0F5ZsVpZx8QTbvnzVMq0Incdd2nLXbe+wE0C1XzU1Qz4
         QFKpvSnTX3PQeQo5Ox2Ug7lIf032fPFxOq0zJA2riZxDMzk82TEGBkWq327XcS3nguGR
         gxVg==
X-Gm-Message-State: APzg51BjmyN+R+rdZQglBa9vjvBFEMi1pHvoJobxqnL2eW4Qgmmojrsd
        kcetZmiYIenqIqBo0vezFYw=
X-Google-Smtp-Source: 
 ANB0VdZdvvgi7gYfFwZte4HA4g1PAh55yLu/hyBeZZRff1RjbPw6E+rk8ZVKkf9CcrXxJt2tfl9lZg==
X-Received: by 2002:a65:52cc:: with SMTP id
 z12-v6mr33124836pgp.69.1536099628609;
        Tue, 04 Sep 2018 15:20:28 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 r19-v6sm22386pgg.39.2018.09.04.15.20.27
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Tue, 04 Sep 2018 15:20:28 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com, liran.alon@oracle.com,
        sean.j.christopherson@intel.com
Cc: tglx@linutronix.de, mingo@redhat.com, x86@kernel.org,
        kvm@vger.kernel.org, Wei Yang <richard.weiyang@gmail.com>
Subject: [PATCH v2] KVM: x86: don't reset root in kvm_mmu_setup()
Date: Wed,  5 Sep 2018 06:20:21 +0800
Message-Id: <20180904222021.2760-1-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Here is the code path which shows kvm_mmu_setup() is invoked after
kvm_mmu_create(). Since kvm_mmu_setup() is only invoked in this code path,
this means the root_hpa and prev_roots are guaranteed to be invalid. And
it is not necessary to reset it again.

    kvm_vm_ioctl_create_vcpu()
        kvm_arch_vcpu_create()
            vmx_create_vcpu()
                kvm_vcpu_init()
                    kvm_arch_vcpu_init()
                        kvm_mmu_create()
        kvm_arch_vcpu_setup()
            kvm_mmu_setup()
                kvm_init_mmu()

This patch set reset_roots to false in kmv_mmu_setup().

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
Reviewed-by: Liran Alon <liran.alon@oracle.com>
---
v2: adjust changelog based on Sean Christopherson comment
---
 arch/x86/kvm/mmu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index a282321329b5..28da8a9bd37b 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -5413,7 +5413,7 @@ void kvm_mmu_setup(struct kvm_vcpu *vcpu)
 {
 	MMU_WARN_ON(VALID_PAGE(vcpu->arch.mmu.root_hpa));
 
-	kvm_init_mmu(vcpu, true);
+	kvm_init_mmu(vcpu, false);
 }
 
 static void kvm_mmu_invalidate_zap_pages_in_memslot(struct kvm *kvm,
